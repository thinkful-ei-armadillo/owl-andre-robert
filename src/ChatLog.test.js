import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatLog from './ChatLog';
import { STORE } from './store';

describe('Testing Chat log', () => {
  // smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatLog></ChatLog>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatLog></ChatLog>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});