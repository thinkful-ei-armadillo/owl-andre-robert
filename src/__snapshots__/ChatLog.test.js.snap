// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Chat log renders the UI as expected 1`] = `
<section
  className="chat-log"
>
  <ul>
    <li
      className="chatEvent"
    >
      <div
        className="message-container"
      >
        <img
          alt="avatar for user"
          src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
        />
        <div>
          <h6>
            Axe Kubicka
          </h6>
           
          <span>
            4:50:46 AM
          </span>
          <p>
            Hello world
          </p>
        </div>
      </div>
    </li>
    <li
      className="chatEvent"
    >
      <span>
        <strong>
          Caty Flucker
        </strong>
         gave a thumbs up.
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span>
        <strong>
          Frank Runciman
        </strong>
         gave a thumbs down.
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span>
        <strong>
          Koren Templeton
        </strong>
         raised their hand.
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span>
        <strong>
          Koren Templeton
        </strong>
         clapped.
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span>
        <strong>
          Koren Templeton
        </strong>
         joined
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span>
        <strong>
          Ashla Attwool
        </strong>
         left
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span>
        <strong>
          Axe Kubicka
        </strong>
         joined the stage
      </span>
    </li>
    <li
      className="chatEvent"
    >
      <span>
        <strong>
          Axe Kubicka
        </strong>
         left the stage
      </span>
    </li>
  </ul>
</section>
`;
